File: changsukuaidi-express-andriod/app/src/main/java/com/changsukuaidi/www/modules/login/LoginModule.java
package com.changsukuaidi.www.modules.login;

import dagger.Module;
import dagger.Provides;

/**
 * @Describe
 * @Author zhouhao
 * @Date 2018/2/7
 * @Contact 605626708@qq.com
 */

@Module
class LoginModule {
    private final LoginFragment mView;

    LoginModule(LoginFragment view) {
        mView = view;
    }

    @Provides
    LoginFragment provideLoginView(){
        return mView;
    }
}

File: WIFIDirect/app/src/main/java/com/thc/www/p2p/module/LoginModule.kt
package com.thc.www.p2p.module

import com.thc.www.p2p.android.login.LoginFragment
import com.thc.www.p2p.annotation.FragmentScope
import dagger.Module
import dagger.Provides

/**
 * @Describe
 * @Author thc
 * @Date  2018/5/7
 */
@Module
class LoginModule(private val view: LoginFragment) {
    @FragmentScope
    @Provides
    fun provideView() = view
}